nntrainer/layers/layer_node.h

```diff
@@ -286,7 +288,10 @@ class LayerNode : public ml::train::Layer, public GraphNode {
 
   RunLayerContext
     run_context; /**< context required for running/execution of the layer. This
-                    will also contain the properties of the layer. */
+                    will also contain the properties of the layer. The
+                    properties will be copied upon final creation. Editing
+                    properties of the layer after init will not the properties
+                    in the context/graph unless intended. */
```

Comment:
I wonder about the purpose of this behavior. My concern is that we could save to the context with `std::any` but we can't save the type.

Reply:
to avoid updating properties once the layer is initialized.

Reply:
> My concern is that we could save to the context with std::any but we can't save the type.

you want to directly save to run_context?

Reply:
How about layer node control this behavior or let it be free for now?

Reply:
let's keep it free for now. how to handle stopping the setProperties() after init() can be handled later.

Reply:
Cool

Reply:
> you want to directly save to run_context?

I mean properties have it's own type. So there is no way to have a member if we are not using std::any. Let's delay the discussion for now :)

Reply:
yes.